tools/tsp-client/src/commands.ts

```diff
@@ -31,6 +33,56 @@ import { sortOpenAPIDocument } from "@azure-tools/typespec-autorest";
 
 const defaultRelativeEmitterPackageJsonPath = joinPaths("eng", "emitter-package.json");
 
+// This function resolves for the correct emitter to use in a language repository based on whether
+// the user has provided an alternate dependency file path or if there are existing configuration
+// files in the repository. Supported configuration files are located at
+// <repo root>/eng/emitter-package.json or <repo root>/eng/tspclientconfig.yaml.
+// The function will return the emitter name and the path to the emitter package.json file relative
+// to repo root. If the default emitter-package.json is used, the path will be undefined.
+async function getEmitter(
+  repoRoot: string,
+  tspConfigData: any, // tspconfig.yaml data
+  globalConfigFile?: TspClientConfig,
+  emitterPackageJsonOverride?: string,
+): Promise<{ emitter: string; path?: string }> {
+  // If an emitter-package.json override value is explicitly provided, use it to get the emitter
+  if (emitterPackageJsonOverride) {
+    return {
+      emitter: await getEmitterFromRepoConfig(emitterPackageJsonOverride),
+      path: relative(repoRoot, emitterPackageJsonOverride),
+    };
+  }
+
+  // If a global config file exists with supportedEmitters configured, use it to
+  // find the right emitter. The list of supported emitters will be checked in order, stopping
+  // at the first match in tspconfig.yaml.
+  if (globalConfigFile && globalConfigFile.supportedEmitters) {
+    for (const supportedEmitter of globalConfigFile.supportedEmitters) {
+      for (const configEmitter of Object.keys(tspConfigData.options) ?? []) {
```

Comment:
nit: is it worth parsing each config into dictionaries to avoid the O(N^2) loop?

Reply:
Done